### ISSUE TITLE
GCC incorrectly detected as Apple Clang on OSX

### ISSUE THREAD

**Issue by dengyaotriangle** [microsoft/vscode-cpptools#6916]
**Type: LanguageService**

**Describe the bug**
- OS and Version: macOS catalina 10.15.7
- VS Code Version: 1.53.0
- C/C++ Extension Version: 1.2.0
- C++ config: Compiler homebrew GCC 10.2, intellisense with c++17 and gcc-x64.
- Error only appear in 1.2.0, any previous version of this extension is ok.
- Intellisense report error while compiler compiles.

**Steps to reproduce**

Code
```cpp
std::filesystem::path x("foo");
std::string y("bar");
x=x/y;
```
yield an error reporting no operator `std::filesystem::__cxx11::path / std::string` exists

**Expected behavior**

No error, since both gcc and clang compiles this code without any warning and error.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this -- it's a regression from 1.1.3 (i.e. downgrading to 1.1.3 is the suggested workaround until we release a fix with 1.2.1).

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.2.1
